### PR TITLE
Extratask

### PR DESCRIPTION
**Pull request by koshik**
Koshik Elena 2539

### CODE REVIEW

**Comment by koshik** [comment]
Я не свой ключ вставляла

**Comment by koshik** [comment]
Sorry for late commit. I have been using API KEY from API explorer and didn't notice it. This key works only one day

**Comment by thevery** [comment]
refresh takes some time, but picture quality is the best so far

**Comment by thevery** [comment]
UX issues: refresh button takes too much space (should be moved to action bar), action bar is useless on the second screen (should be removed)

**Comment by thevery** [comment]
app is ok (20-2=18), code review pending

**Comment by thevery** [comment]
toast to started progress is nice!

**Comment by thevery** [comment]
often crashes with

E/AndroidRuntime( 8752): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.extratask/com.example.extratask.PictureActivity}: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
E/AndroidRuntime( 8752):    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2195)
E/AndroidRuntime( 8752):    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
E/AndroidRuntime( 8752):    at android.app.ActivityThread.access$800(ActivityThread.java:135)
E/AndroidRuntime( 8752):    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
E/AndroidRuntime( 8752):    at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 8752):    at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime( 8752):    at android.app.ActivityThread.main(ActivityThread.java:5017)
E/AndroidRuntime( 8752):    at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 8752):    at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime( 8752):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
E/AndroidRuntime( 8752):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
E/AndroidRuntime( 8752):    at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime( 8752): Caused by: android.database.CursorIndexOutOfBoundsException: Index 0 requested, with a size of 0
E/AndroidRuntime( 8752):    at android.database.AbstractCursor.checkPosition(AbstractCursor.java:426)
E/AndroidRuntime( 8752):    at android.database.AbstractWindowedCursor.checkPosition(AbstractWindowedCursor.java:136)
E/AndroidRuntime( 8752):    at android.database.AbstractWindowedCursor.getBlob(AbstractWindowedCursor.java:44)
E/AndroidRuntime( 8752):    at com.example.extratask.DataBase.getImage(DataBase.java:92)
E/AndroidRuntime( 8752):    at com.example.extratask.PictureActivity.onCreate(PictureActivity.java:22)
E/AndroidRuntime( 8752):    at android.app.Activity.performCreate(Activity.java:5231)
E/AndroidRuntime( 8752):    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
E/AndroidRuntime( 8752):    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
E/AndroidRuntime( 8752):    ... 11 more

File: AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.example.extratask"
```
Comment (thevery) [inline AndroidManifest.xml:3]: bad package

**Comment by thevery** [comment]
icon is too big

File: res/layout/grid_item.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/grid_item.xml:3]: extra layout seems to be useless

File: res/layout/main.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <Button
+            android:layout_width="fill_parent"
+            android:layout_height="wrap_content"
+            android:text="@string/ref"
+            android:onClick="refresh"
+            android:id="@+id/button" android:layout_gravity="center"/>
+    <GridView
+            android:id="@+id/gridview"
+            android:layout_width="fill_parent"
+            android:layout_height="fill_parent"
```
Comment (thevery) [inline res/layout/main.xml:18]: should be 0dp with weight=1

File: res/layout/picture.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (thevery) [inline res/layout/picture.xml:3]: useless layout

File: src/com/example/extratask/DownloadService.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        try {
+            URL imageUrl = new URL(url);
+            URLConnection ucon = imageUrl.openConnection();
+
+            InputStream is = ucon.getInputStream();
+            BufferedInputStream bis = new BufferedInputStream(is);
+
+            ByteArrayBuffer baf = new ByteArrayBuffer(500);
+            int current = 0;
+            while ((current = bis.read()) != -1) {
```
Comment (thevery) [inline src/com/example/extratask/DownloadService.java:78]: you should not read by 1 char, read to buffer instead

File: src/com/example/extratask/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        gridView.setOnItemClickListener(gridviewOnItemClickListener);
+
+    }
+
+    @Override
+    public void onReceiveResult(int resultCode, Bundle data) {
+        db.getAllImages(images);
+        adapter.notifyDataSetChanged();
+        Toast toast = Toast.makeText(getApplicationContext(),
+                "Download complete.", Toast.LENGTH_SHORT);
```
Comment (thevery) [inline src/com/example/extratask/MainActivity.java:85]: strings.xml

**Comment by thevery** [comment]
20-2-4=14

**Pull request closed without merging**